frontend/src/features/workspace/WorkspacePage.tsx

```diff
@@ -17,7 +24,28 @@ import { WorkspaceProvider } from './WorkspaceProvider'
 export const CONTAINER_MAXW = '69.5rem'
 
 export const WorkspacePage = (): JSX.Element => {
-  const { data: { siteBannerContent, adminBannerContent } = {} } = useEnv()
+  const {
+    data: {
+      siteBannerContent,
+      adminBannerContent,
+      adminFeedbackDisplayFrequency,
+    } = {},
+  } = useEnv()
+  const { user, isLoading } = useUser()
+  const [isDisplayFeedback, setIsDisplayFeedback] = useState(false)
+  const isMobile = useIsMobile()
+
+  const adminFeedbackKey = useMemo(() => {
+    return ADMIN_FEEDBACK_HISTORY_PREFIX + user?._id
+  }, [user])
+
+  const [lastFeedbackTime, setLastFeedbackTime] =
+    useLocalStorage<number>(adminFeedbackKey)
+  const [isAdminFeedbackEligible, setIsAdminFeedbackEligible] =
+    useSessionStorage<boolean>(ADMIN_FEEDBACK_SESSION_KEY, false)
+
+  // Memo current time on page load to prevent re-renders from update to current time
+  const currentTime = useMemo(() => Date.now(), [])
```

Comment:
Im thinking this should be `useRef` as it better represents the reason why we don't want to regenerate this data.

`useMemo` implies that we want to memoize the function as the inner function is computationally expensive.

`useRef` implies that this value should not change over the lifetime of this component.

Reply:
Oh wow thanks Ken! [384d3d9](https://github.com/opengovsg/FormSG/pull/6465/commits/384d3d9dd3f45dd2ad7b37ae1442b86e2966fb56)

And thanks for explaining the difference in usage of the two hooks! 💯